### ISSUE TITLE
dissemination smartsky plugins renku

### ISSUE THREAD

**Issue by volodymyrss** [oda-hub/hugo-odahub#70]
two plugins:
* graph visualization https://github.com/oda-hub/renku-graph-vis/ 
* aqs plugin explaining how we detect connection to external resources and impact on reproducibility

who maintains https://gitlab.renkulab.io/learn-renku @rokroskar ? 

can make a discourse post renku.discourse.group , we post it, share draft in advance.

put own social media (linkedin, mastodon)

series of renku blog posts https://medium.com/the-renku-blog can add a post there, we need to contact Elisabeth (find community manager email)

we can add a note/reference https://renku.readthedocs.io/en/stable/renku-python/docs/plugins.html  @burnout87 can do it

we have some manual here https://odahub.io/docs/guide-development/

try to disseminate in unige [DSCS](https://datascience.unige.ch/en/research/uniges-data-science-days)

**Comment by burnout87** [comment]
The link on the page https://renku.readthedocs.io/en/stable/renku-python/docs/plugins.html is not working, probably outdated

**Comment by rokroskar** [comment]
Right... that's because the renku-python docs are embedded in the other repo 🙈 

Here's the correct place: https://github.com/SwissDataScienceCenter/renku-python/blob/develop/docs/plugins.rst

**Comment by burnout87** [comment]
Thanks

**Comment by burnout87** [comment]
Changes with notes on the two plugins have been merged

https://github.com/SwissDataScienceCenter/renku-python/blob/develop/docs/plugins.rst has been merged

**Comment by burnout87** [comment]
A first draft of the post about the plugins.

* What do you think?
  * Is it too focused on the workflow-development aspect?
  * should it be a bit more general or a but more technical?
  * is too detailed ? or is it too short?

Also, not sure this is the best place for talking about it, feel free to suggest a more appropriate location.

---

## Support the workflow development via renku plugin

To support the development of workflows in Renku, a set of dedicated functionalities, provided as Renku plugins, are made available. Specifically, these plugins are:

* `renku-graph-vis`: to offer a visualization of the project's Knowledge Graph.
* `renku-aqs-annotation`: to Intercept calls to astroquery functions and store them in the project's Knowledge Graph.

### `renku-graph-vis`

This plugin provides a graphical representation of the renku repository's knowledge graph in two ways:

* From within the renkulab session
* Via a set of CLI commands. 
 
In this way, the ability to have a live overview of the ongoing development for real-time monitoring is introduced.
The animation below shows the automatic updates of the graph during a Renkulab session, with information about the execution of a notebook (`papermill` command), upon its completion.

![image](https://github.com/oda-hub/hugo-odahub/assets/3757833/757682a8-fbdd-486d-879d-5bc3d0088502)

To initiate or access the live graph visualization during your session, simply click on the Graph icon located on the main page, as shown in the image below.

![image](https://github.com/oda-hub/hugo-odahub/assets/3757833/05950171-994d-4257-9191-33de0a06fc03)

As mentioned above, two CLI commands are also offered:

* `display`: to generate a representation of the graph over an output image.
* `show-graph`: to start an interactive visualization of the graph over the browser.

More technical details are presented in the README of the repo page: https://github.com/oda-hub/renku-graph-vis/

### `renku-aqs-annotation`

This plugin enables the tracking of the accesses to astronomical archives and services in the project Knowledge Graph.
In particular, the plugin intercepts several key [`astroquery`](https://astroquery.readthedocs.io/en/latest/index.html) methods and stores annotations containing information about the calls to these methods (like the arguments used in the call itself) to the project's Knowledge Graph.

In the image below, the graph produced by the `renku-graph-vis` plugin during a renkulab session, exposes the information added to the project Knowledge Graph as a result of the interception of the `astroquery` methods. Specifically, it can be seen that during a `papermill` run of the `test-notebook.ipynb` notebook (that produced `out.ipynb` as an output notebook) a call to the `astroquery` method `query_object`, using the `SimbadClass`, has been detected. This notebook is requesting the object `Mrk 421` object. The highlighted labels on the edges provide information about the relationship between the two nodes: during the `papermill`execution, a call to the `query_object` method is executed (call label) and in turn, this requests the Astrophysical Object `Mrk 421`.

![image](https://github.com/oda-hub/hugo-odahub/assets/3757833/70fbb6be-c6db-413b-98ed-c0b9ac24d9e5)

More technical details can be found in the `README` of the repo page: https://github.com/oda-hub/renku-aqs-annotation

**Comment by burnout87** [comment]
First version of two short video demo.

* installation

https://github.com/oda-hub/hugo-odahub/assets/3757833/a9b2db20-0ff7-477d-8ec5-2be3a0ac9373



* demo


https://github.com/oda-hub/hugo-odahub/assets/3757833/b4995bb3-5532-46e5-8084-e14865a7e4de

**Comment by volodymyrss** [comment]
> First version of two short video demo.
> 
>     * installation
> 
> 
> demo_video_plugins_installation.mp4
> 
>     * demo
> 
> 
> demo_video_plugins_usage.mp4

Nice! But we are waiting for some 1.5min for renku session start. Could you please fast-forward them?

**Comment by burnout87** [comment]
> 
> Nice! But we are waiting for some 1.5min for renku session start. Could you please fast-forward them?

I did a 10x increase in speed of that part of video, what do you think?



https://github.com/oda-hub/hugo-odahub/assets/3757833/82d99754-07cb-4627-bc1f-d17705e85a5a

**Comment by volodymyrss** [comment]
> > Nice! But we are waiting for some 1.5min for renku session start. Could you please fast-forward them?
> 
> I did a 10x increase in speed of that part of video, what do you think?
> demo_video_plugins_usage_faster.mp4

It's not a bad way! Could you maybe add some ">>>" label when acceleration happens?
Also there are other a bit slow parts.

**Comment by burnout87** [comment]
I added a "fast-forward" image/logo when speeding up the vide...What do you think? @volodymyrss 

https://github.com/oda-hub/hugo-odahub/assets/3757833/a5713518-507e-49aa-ad61-a371b5064b07

**Comment by volodymyrss** [comment]
> I added a "fast-forward" image/logo when speeding up the vide...What do you think? @volodymyrss
> demo_video_plugins_usage_faster_labelled.mp4

The fast-forward looks good I think, thanks.

**Comment by burnout87** [comment]
I made a faster version also of the installation video

https://github.com/oda-hub/hugo-odahub/assets/3757833/093c0ac1-2bd1-427a-b600-b1f573b30b80

**Comment by burnout87** [comment]
> 
>     * [ ]  series of renku blog posts https://blog.renkulab.io/ https://medium.com/the-renku-blog https://github.com/oda-hub/renku-blog can add a post there in our fork, need to mention @rokroskar

https://github.com/SwissDataScienceCenter/renku-blog/pull/17
